ambry-api/src/main/java/com/github/ambry/config/CloudConfig.java

```diff
@@ -465,7 +465,7 @@ public class CloudConfig {
   public static final double DEFAULT_BACKUP_NODE_CPU_SCALE = -1; // 1 thread
   // num_cpu-1 threads, prevent a fork bomb
   public static final double MIN_BACKUP_NODE_CPU_SCALE = 1 - Double.valueOf(Runtime.getRuntime().availableProcessors());
-  public static final double MAX_BACKUP_NODE_CPU_SCALE = 4.0; // 4 * num_cpus, prevent a fork bomb
+  public static final double MAX_BACKUP_NODE_CPU_SCALE = 16.0; // 16 * num_cpus, this is max, not actual config value
```

Comment:
What does scale mean here? Are we scaling the number of threads by 16 per CPU?

Reply:
This patch increase the ceiling on the number of worker threads, not the actual number of worker threads itself. That will set in the internal fork.

Number of threads on the host = scale * num_cpus

scale lies between min and max values. This patch moves up the max value.

Reply:
Makes sense, thank you for explaining